### PR TITLE
Introduce bounds to search for sparkles!

### PR DESCRIPTION
**Pull request by dizzyd**
In c368692a3, the recursive search was changed to iterative. The big problem with both approaches, however is that as more and more positions are scanned, it takes longer to actually determine if the position was scanned. This PR introduces a radius which is used to get a fixed set of block positions to scan and as each position is scanned, particles are generated. This reduces memory footprint and avoids a O(n^2) search.

Not quite working yet; pending me not being on a plane. :)

### CODE REVIEW

**Comment by thiakil** [comment]
Have you measured performance from checking the already iterated set before adding to the queue? (would help reduce at least some of the memory impact in this case)

File: src/main/java/mekanism/client/SparkleAnimation.java [hunk 1]
```diff
@@ -16,59 +18,69 @@
...
+                      pos.getY() + random.nextDouble(), pos.getZ() + 1.01, 0, 0, 0);
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + -.01,
+                      pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + 1.01,
+                      pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
+            }
+        }
+
+        long elapsed = System.nanoTime() - start;
+        Mekanism.logger.info("Generated {} sparkles in {} ms", sparkles, elapsed / 1000000d);
```
Comment (thommy101) [inline src/main/java/mekanism/client/SparkleAnimation.java]: Is this still some debug info?

Comment (dizzyd) [reply src/main/java/mekanism/client/SparkleAnimation.java]: Yes - I left that in there while the review is ongoing. Will remove before final commit.

**Comment by dizzyd** [comment]
In reference to your comment @thiakil - it was already checking if a coordinate was in the set, but it was having to check n^2 times if it was in the set, due to the traversal. The lookup wasn't _that_ expensive, but that coupled with memory allocation and the creation of a thread could push total generation time into 10s of milliseconds at times. With this setup, it's reliably 1ms or less and we avoid memory allocation and thread creation.

**Pull request merged**